plugins/chat/spec/system/transcript_spec.rb

```diff
@@ -0,0 +1,246 @@
+# frozen_string_literal: true
+
+RSpec.describe "Quoting chat message transcripts", type: :system, js: true do
+  fab!(:current_user) { Fabricate(:user) }
+  fab!(:chat_channel_1) { Fabricate(:chat_channel) }
+  fab!(:chat_message_1) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:chat_message_2) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:chat_message_3) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:chat_message_4) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:chat_message_5) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:topic) { Fabricate(:with_posts_topic) }
+  let(:last_topic_post) { topic.posts.last }
+
+  let(:chat_page) { PageObjects::Pages::Chat.new }
+  let(:chat_channel_page) { PageObjects::Pages::ChatChannel.new }
+  let(:topic_page) { PageObjects::Pages::Topic.new }
+
+  before do
+    chat_system_bootstrap(current_user, [chat_channel_1])
+    sign_in(current_user)
+  end
+
+  def select_message(message, mobile: false)
+    if page.has_css?(".chat-message-container.selecting-messages")
+      chat_channel_page.message_by_id(message.id).find(".chat-message-selector").click
+    else
+      # we long press instead of hover on mobile
+      if mobile
+        chat_channel_page.message_by_id(message.id).click(delay: 0.5)
+      else
+        chat_channel_page.message_by_id(message.id).hover
+      end
+
+      # we also have a different actions menu on mobile
+      if mobile
+        find(".chat-message-action-item[data-id=\"selectMessage\"]").click
+      else
+        expect(page).to have_css(".chat-message-actions .more-buttons")
+        find(".chat-message-actions .more-buttons").click
+        find(".select-kit-row[data-value=\"selectMessage\"]").click
+      end
+    end
+  end
+
+  def cdp_allow_clipboard_access!
```

Comment:
Candidate to move into a helper after Joffrey's PR. Actually think all these CDP related ones should be in a single class as self methods.